index.bs

```diff
@@ -277,7 +277,94 @@ request.onsuccess = function() {
   db = request.result; // db.version will be 3.
 };
 </pre>
+</aside>
+
+<aside class=example id=handling-versionchange>
+A single database can be used by multiple clients (pages and workers)
+simultaneously - transactions ensure they don't clash while reading and writing.
+However, if a new client wants to upgrade the database (via the "upgradeneeded"
+event), it cannot do so until all other clients close their connection to the
+current version of the database.
+
+To avoid blocking a new client from upgrading, clients should listen for the
+"versionchange" event. This fires when another client is wanting to upgrade the
+database. To allow this to continue, react to the "versionchange" event by doing
+something that ultimately closes this client's connection to the database.
+
+One way of doing this is to reload the page:
+
+<pre class=lang-javascript>
+db.onversionchange = function() {
+  // First, save any unsaved data:
+  saveUnsavedData().then(function() {
+    // If the document isn't being actively used, it may be appropriate to reload
+    // the page without the user's interaction.
+    if (!document.hasFocus()) {
+      location.reload();
+      // Reloading will close the database, and also reload with the new JavaScript
+      // and database definitions.
+    } else {
+      // If the document has focus, it may be too disruptive to reload the page.
+      // Maybe ask the user to do it manually:
+      displayMessage("Please reload this page for the latest version.");
+    }
+  });
+};
+
+function saveUnsavedData() {
+  // How you do this depends on your app.
+}
+
+function displayMessage() {
+  // Show a non-modal message to the user.
+}
+</pre>
+
+Another way is to call the database's close method. Other code on your page
+should react to the database's "close" event, so it knows the database is no
```

Comment:
And therefore it might be worth adding a note to the `onclose` definition that it's only fired in abnormal close scenarios. Look for the _forced close_ flag.